### PR TITLE
Pulls test setup from showcase/references to showcase/

### PR DESCRIPTION
**Pull request by piotrswigon**

### CODE REVIEW

File: javatests/arcs/showcase/ShowcaseEnvironment.kt [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+import org.junit.runners.model.Statement
+import kotlin.coroutines.EmptyCoroutineContext
+
+/**
+ * A JUnit rule setting up an Arcs environment for showcasing features.
+ *
+ * Usage example follows:
+ *
+ * ```
+ * @get:Rule val env = ShowcaseEnvironment(
```
Comment (jblebrun) [inline javatests/arcs/showcase/ShowcaseEnvironment.kt:41]: nice!!

File: javatests/arcs/showcase/ShowcaseEnvironment.kt [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+ *   // Do something with it.
+ *   env.getParticle<SomeParticle>(arc).query("what's the answer?")
+ *
+ *   // Assert result.
+ *   assertThat(env.getParticle<OtherParticle>(arc).state).isEqualTo(42)
+ * }
+ * ```
+ */
+@ExperimentalCoroutinesApi
+class ShowcaseEnvironment(
```
Comment (jblebrun) [inline javatests/arcs/showcase/ShowcaseEnvironment.kt:60]: Is there any convention about always naming `TestRule` classes with a `Rule` suffix?

File: javatests/arcs/showcase/references/Arcs.kt [hunk 3]
```diff
@@ -1,176 +0,0 @@
...
-import arcs.sdk.android.storage.ServiceStoreFactory
-import arcs.sdk.android.storage.service.ConnectionFactory
-import kotlinx.coroutines.ExperimentalCoroutinesApi
-import kotlinx.coroutines.runBlocking
-import kotlinx.coroutines.withTimeout
-import kotlin.coroutines.EmptyCoroutineContext
-
-/** Container for WriteRecipe specific things */
-@ExperimentalCoroutinesApi
-class ArcsStorage(private val arcs: Arcs) {
```
Comment (jblebrun) [inline javatests/arcs/showcase/references/Arcs.kt:32]: I would like to keep this DAO-like object wrapper if possible...

File: javatests/arcs/showcase/references/ReadWriteTest.kt [hunk 4]
```diff
@@ -1,83 +1,52 @@
...
-            // Ignoring this exception should be OK.
-            try {
-                dbManager.resetAll()
-            } catch(e: Exception) {
-                println("Ignoring dbManager.resetAll() exception: $e")
-            }
-
-        }
-    }
+    private val storage = ArcsStorage(env)
```
Comment (jblebrun) [inline javatests/arcs/showcase/references/ReadWriteTest.kt:27]: thanks!

File: javatests/arcs/showcase/ShowcaseEnvironment.kt (hunk 2 above)
Comment (piotrswigon) [reply javatests/arcs/showcase/ShowcaseEnvironment.kt:60]: I don't think so, based on this: https://junit.org/junit4/javadoc/4.12/org/junit/rules/package-summary.html

You have: TemporaryFolder, ErrorCollector, Stopwatch etc.

File: javatests/arcs/showcase/references/Arcs.kt (hunk 3 above)
Comment (piotrswigon) [reply javatests/arcs/showcase/references/Arcs.kt:32]: Reverted!

**Pull request merged**